### DISCUSSION TITLE
Is it possible to access location.state from a clientLoader?

### DISCUSSION THREAD

**Discussion by lockr7** [remix-run/react-router#11330] [Q&A] [answered]
I saw https://github.com/remix-run/remix/discussions/8243 which makes sense but now that clientLoaders exist, curious if there's a way to access this

**Comment by sergiodxa** [comment, marked as answer]
Using `window.history.state` in a clientLoader or clientAction should be possible to access the `location.state`

**Reply by misha-erm** [reply]
@sergiodxa I found that window.history.state lags behind. After I `useNavigate` with state window.history.state inside loader misses the properties I set. But they are present if I check state with `useLocation`. Is it something expected or should I dig deeper and create the issue?

**Reply by jbojcic1** [reply]
@misha-erm were you ever able to solve that? I have similar issue described here https://github.com/remix-run/remix/discussions/10721

**Reply by misha-erm** [reply]
unfortunately I don't remember :( 
I think I implemented my feature in a different way and switched to "global" state variable, e.g.

```
feed-positions.tsx


const feedPositions: Map<string, GridStateSnapshot> = new Map();

export const hasGridFeedPosition = (key: string) => feedPositions.has(key);
export const removeGridFeedPosition = (key: string) =>
  feedPositions.delete(key);
  
export const setFeedPosition = (gss: GridStateSnapshot) => {
    feedPositions.set(key, gss);
  };

```

and inside loader:

```
const clientLoader = () => {
  if (hasFeedPosition(history.state.key)) {
    ...
  } 
}
```

**Reply by misha-erm** [reply]
@jbojcic1 not sure but maybe you should use HashRouter if you need your loaders to be triggered on hash change
